backend/level_slice.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2020 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+"""A Module for processing single slice."""
+from collections import defaultdict
+from math import ceil
+
+from downsample import strategy_reducer
+from utils import convert_to_csv
+from utils import parse_csv_line
+
+
+class LevelSlice:
+    """A class for processing slice and its records."""
```

Comment:
Explained in MultipleLevelPreprocess, because I add a class for metadata, and this way the explanations can cover the definitions of metadata, slice and raw data.